pkg/pillar/utils/file/file.go

```diff
@@ -14,11 +14,45 @@ import (
 	"time"
 
 	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/sirupsen/logrus"
 	"golang.org/x/sys/unix"
 )
 
 const maxCounterReadSize = 16384 // Max size of counter file
 
+func dirSync(dirName string) error {
+	f, err := os.OpenFile(dirName, os.O_RDONLY, 0755)
+	if err != nil {
+		return err
+	}
+
+	err = f.Sync()
+	if err != nil {
+		return err
+	}
+
+	err = f.Close()
+	return err
+}
+
+func backupFile(fileName string) error {
+	_, err := os.Stat(fileName)
+	if err != nil {
+		// File doesn't exist
+		return err
+	}
+
+	bacName := fmt.Sprintf("%s.bac", fileName)
```

Comment:
Nit but I'm used to this being spelled .bak. Never seen .bac before.

Reply:
Updated:
```
~# ls /hostfs/persist/status/nim/DevicePortConfigList/
global.json      global.json.bak
```